src/main/java/com/wolox/training/controllers/BookController.java

```diff
@@ -119,4 +124,18 @@ public Book updateBook(
     bookRepository.findById(id).orElseThrow(() -> new NotFoundException(Constant.BOOK_NOT_FOUND));
     return bookRepository.save(book);
   }
+
+  @ApiOperation(value = "Given an ISBN, the corresponding book is returned", response = Book.class)
+  @ApiResponses(value = {
+      @ApiResponse(code = 200, message = Constant.OK_MESSAGE),
+      @ApiResponse(code = 404, message = Constant.NOT_FOUND_MESSAGE),
+      @ApiResponse(code = 401, message = Constant.NOT_AUTHORIZED_MESSAGE),
+      @ApiResponse(code = 403, message = Constant.FORBIDDEN_MESSAGE)
+  })
+  @GetMapping("/isbn/{ISBN}")
+  public Book findByISBN(
+      @ApiParam(value = "ISBN of the book to be found", required = true) @PathVariable String ISBN)
+      throws JsonProcessingException {
+    return openLibraryService.bookInfo(ISBN);
```

Comment:
We should be able to response with different status code base on the result operation. Check the card description